backend/handlertest/handlertest.go

```diff
@@ -0,0 +1,271 @@
+
+// Handler a test handler implementing backend.Handler.
+type Handler struct {
+	QueryDataFunc         backend.QueryDataHandlerFunc
+	CallResourceFunc      backend.CallResourceHandlerFunc
+	CheckHealthFunc       backend.CheckHealthHandlerFunc
+	CollectMetricsFunc    backend.CollectMetricsHandlerFunc
+	SubscribeStreamFunc   backend.SubscribeStreamHandlerFunc
+	PublishStreamFunc     backend.PublishStreamHandlerFunc
+	RunStreamFunc         backend.RunStreamHandlerFunc
+	MutateAdmissionFunc   backend.MutateAdmissionFunc
+	ValidateAdmissionFunc backend.ValidateAdmissionFunc
+	ConvertObjectsFunc    backend.ConvertObjectsFunc
+}
+
+func (h Handler) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	if h.QueryDataFunc != nil {
+		return h.QueryDataFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+func (h Handler) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
+	if h.CallResourceFunc != nil {
+		return h.CallResourceFunc(ctx, req, sender)
+	}
+
+	return nil
+}
+
+func (h Handler) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	if h.CheckHealthFunc != nil {
+		return h.CheckHealthFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+func (h Handler) CollectMetrics(ctx context.Context, req *backend.CollectMetricsRequest) (*backend.CollectMetricsResult, error) {
+	if h.CollectMetricsFunc != nil {
+		return h.CollectMetricsFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+func (h Handler) SubscribeStream(ctx context.Context, req *backend.SubscribeStreamRequest) (*backend.SubscribeStreamResponse, error) {
+	if h.SubscribeStreamFunc != nil {
+		return h.SubscribeStreamFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+func (h Handler) PublishStream(ctx context.Context, req *backend.PublishStreamRequest) (*backend.PublishStreamResponse, error) {
+	if h.PublishStreamFunc != nil {
+		return h.PublishStreamFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+func (h Handler) RunStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
+	if h.RunStreamFunc != nil {
+		return h.RunStreamFunc(ctx, req, sender)
+	}
+
+	return nil
+}
+
+func (h Handler) ValidateAdmission(ctx context.Context, req *backend.AdmissionRequest) (*backend.ValidationResponse, error) {
+	if h.ValidateAdmissionFunc != nil {
+		return h.ValidateAdmissionFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+func (h Handler) MutateAdmission(ctx context.Context, req *backend.AdmissionRequest) (*backend.MutationResponse, error) {
+	if h.MutateAdmissionFunc != nil {
+		return h.MutateAdmissionFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+func (h Handler) ConvertObjects(ctx context.Context, req *backend.ConversionRequest) (*backend.ConversionResponse, error) {
+	if h.ConvertObjectsFunc != nil {
+		return h.ConvertObjectsFunc(ctx, req)
+	}
+
+	return nil, nil
+}
+
+type HandlerMiddlewareTest struct {
```

Comment:
Nit: Since already in a "test" package, would be inclined to just call this:
```suggestion
type HandlerMiddleware struct {
```

Reply:
It was like this earlier so rather keep it if fine with you?